From b216cb0740b6e0107509145cadd0671fda62e89c Mon Sep 17 00:00:00 2001
From: Bence Balogh <bence.balogh@arm.com>
Date: Tue, 28 Nov 2023 15:33:12 +0100
Subject: [PATCH 9/9] Fix psa_ipc service's psa_call

The wrong parameter was passed to the psa_ipc_phys_to_virt()
function which resulted in faulty behavior.

Upstream-Status: Submitted [https://review.trustedfirmware.org/c/TS/trusted-services/+/27858]
Signed-off-by: Bence Balogh <bence.balogh@arm.com>
---
 components/rpc/psa_ipc/service_psa_ipc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/components/rpc/psa_ipc/service_psa_ipc.c b/components/rpc/psa_ipc/service_psa_ipc.c
index 36c8e367..4bf86716 100644
--- a/components/rpc/psa_ipc/service_psa_ipc.c
+++ b/components/rpc/psa_ipc/service_psa_ipc.c
@@ -176,13 +176,13 @@ static psa_status_t __psa_call(struct rpc_caller_interface *caller, psa_handle_t
 	if (!resp_msg || !out_len || resp_msg->reply != PSA_SUCCESS)
 		goto caller_end;
 
-	out_vec_param = (struct psa_outvec *)psa_ipc_phys_to_virt(caller,
+	out_vec_param = (struct psa_outvec *)psa_ipc_phys_to_virt(caller->context,
 				psa_u32_to_ptr(resp_msg->params.out_vec));
 
 	for (i = 0; i < resp_msg->params.out_len; i++) {
 		out_vec[i].len = out_vec_param[i].len;
 		unaligned_memcpy(psa_u32_to_ptr(out_vec[i].base),
-				 psa_ipc_phys_to_virt(caller,
+				 psa_ipc_phys_to_virt(caller->context,
 				      psa_u32_to_ptr(out_vec_param[i].base)),
 				 out_vec[i].len);
 	}
-- 
2.25.1

